src/pint/toa.py

```diff
@@ -1639,6 +1639,37 @@ def get_flag_value(self, flag, fill_value=None, as_type=None):
             result.append(val)
         return result, valid_index
 
+    def set_flag_values(self, flagname, flagvalues):
+        """Set the requested TOA flag values to TOAs
+
+        Parameters
+        ----------
+        flagname : str
+            Name of flag to add/modify
+        flagvalues : str, int, float, or np.ndarray
+            Scalar or array of values.  If array, should match length of TOAs.  All values are converted to string before storing.
+        """
+        assert np.isscalar(flagvalues) or (len(flagvalues) == self.ntoas)
+        for ii in range(self.ntoas):
+            if np.isscalar(flagvalues):
+                if not np.isnan(flagvalues):
+                    self.table["flags"][ii][flagname] = str(flagvalues)
+            else:
+                if not np.isnan(flagvalues[ii]):
+                    self.table["flags"][ii][flagname] = str(flagvalues[ii])
+
+    def delete_flag(self, flagname):
```

Comment:
Can this already be done by assigning `None` or `""` to the flag?

Reply:
Will that actually delete it or just set it to a null value?  I think the later, and there may be cases where the former is desired.  Certainly that was done at several places in the code so I just provided a wrapper.

Reply:
I'm pretty sure I arranged for flags set to a null value to be deleted, since there is no way to represent that in a tim file.